dbms/src/Flash/Planner/Plans/PhysicalExpand.cpp

```diff
@@ -0,0 +1,119 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/FailPoint.h>
+#include <Common/Logger.h>
+#include <Common/TiFlashException.h>
+#include <DataStreams/ExpressionBlockInputStream.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <Flash/Coprocessor/DAGExpressionAnalyzer.h>
+#include <Flash/Coprocessor/DAGPipeline.h>
+#include <Flash/Pipeline/Exec/PipelineExecBuilder.h>
+#include <Flash/Planner/FinalizeHelper.h>
+#include <Flash/Planner/PhysicalPlanHelper.h>
+#include <Flash/Planner/Plans/PhysicalExpand.h>
+#include <Interpreters/Context.h>
+#include <Operators/ExpressionTransformOp.h>
+#include <Operators/FilterTransformOp.h>
+#include <fmt/format.h>
+
+namespace DB
+{
+PhysicalPlanNodePtr PhysicalExpand::build(
+    const Context & context,
+    const String & executor_id,
+    const LoggerPtr & log,
+    const tipb::Expand & expand,
+    const PhysicalPlanNodePtr & child)
+{
+    assert(child);
+
+    child->finalize();
+
```

Comment:
`child->finalize(expand_actions->getRequiredColumns());` in `PhysicalExpand::finalize` is enought.
```suggestion

```